storage/src/system/auction/detail.rs

```diff
@@ -896,28 +896,38 @@ where
     }
 }
 
-pub fn seigniorage_recipient<P>(
+pub fn seigniorage_recipients<P>(
     provider: &mut P,
-    validator_bid: &ValidatorBid,
-) -> Result<SeigniorageRecipient, Error>
+    validator_weights: &ValidatorWeights,
+    validator_bids: &ValidatorBids,
+) -> Result<SeigniorageRecipients, Error>
 where
     P: RuntimeProvider + ?Sized + StorageProvider,
 {
-    let mut delegator_stake: BTreeMap<PublicKey, U512> = BTreeMap::new();
-    for delegator_bid in read_delegator_bids(provider, validator_bid.validator_public_key())? {
-        if delegator_bid.staked_amount().is_zero() {
-            continue;
+    let mut recipients = SeigniorageRecipients::new();
+    for (validator_public_key, validator_stake) in validator_weights {
+        let validator_bid = validator_bids
+            .get(validator_public_key)
+            .ok_or(Error::ValidatorNotFound)?;
+        let mut delegator_stake: BTreeMap<PublicKey, U512> = BTreeMap::new();
+        for delegator_bid in read_delegator_bids(provider, validator_public_key)? {
+            if delegator_bid.staked_amount().is_zero() {
+                continue;
+            }
+            delegator_stake.insert(
+                delegator_bid.delegator_public_key().clone(),
+                delegator_bid.staked_amount(),
+            );
         }
-        delegator_stake.insert(
-            delegator_bid.delegator_public_key().clone(),
-            delegator_bid.staked_amount(),
+
+        let seigniorage_recipient = SeigniorageRecipient::new(
+            *validator_stake,
```

Comment:
I think there's a problem here - this `validator_stake` will include not only the validator's credit (which is what we want), but also the delegators' stakes (which is wrong), as it is calculated using the `total_staked_amount` function [here](https://github.com/casper-network/casper-node/pull/4759/files#diff-a0327a7fb4802880b428f79ca1fb4627e720a4d86bb02e3c4e10fa86663fd9d5L136). The stake here should be just the amount staked by the validator + the credit, I believe.

Reply:
I agree, I think the easiest way around it would be to not destructure the bids at all and calculate the validator stake (+ the credit) per validator key.

Reply:
addressed